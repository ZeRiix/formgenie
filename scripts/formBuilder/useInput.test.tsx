import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import { useInput } from "./useInput";

function InputTest(
	props: {
		label: string;
		value: string;
		onChange(value: any): void;
		placeholder: string;
		tt: string;
	},
) {
	return (
		<>
			<input
				onChange={(event) => void props.onChange(event.target.value)}
				value={props.value}
				placeholder={props.placeholder}
				name={props.label}
			/>
			<p>{props.tt}</p>
		</>
	);
}

describe("useInput", () => {
	it("should return the correct template", () => {
		const { Template } = useInput(InputTest, {
			label: "test",
			defaultValue: "default",
			props: {
				placeholder: "placeholder",
				tt: "tt",
			},
		});

		const componentRender = render(
			<Template
				label="test"
				value="default"
				onChange={() => void 0}
				errorMessage="error"
			/>,
		);
		expect(componentRender.asFragment()).toMatchSnapshot();
	});
});
