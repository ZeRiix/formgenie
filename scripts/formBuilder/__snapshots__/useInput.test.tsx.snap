// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useInput > should return the correct template 1`] = `
<DocumentFragment>
  <label
    for="test"
  >
    test
  </label>
  <input
    name="test"
    placeholder="placeholder"
    value="default"
  />
  <p>
    tt
  </p>
  <span>
    error
  </span>
</DocumentFragment>
`;
